File: java/P1.java
// project euler #1 in java by antekone
// http://anadoxin.org/blog
//
// usage:
//
// $ javac P1.java
// $ java P1

public class P1 {
    public static void main(String[] args) {
        int sum = 0;

        for(int i = 0; i < 1000; i++) {
            if(i % 3 == 0 || i % 5 == 0) {
                sum += i;
            }
        }

        System.out.println(sum);
    }
}

File: erlang/p1.erl
% project euler #1 in Erlang by antekone
% http://anadoxin.org/blog
%
% usage:
%
% $ erl -compile p1.erl && erl -noshell -s p1 problem1

-module(p1).
-export([problem1/0]).

problem1_gen(UpTo) -> lists:foldl(fun(Z, Sum) ->
                                      Z + Sum
                              end, 0, [X || X <- lists:seq(0, UpTo), (X rem 3 =:= 0) or (X rem 5 =:= 0)]).

problem1() -> io:fwrite("~p\n", [problem1_gen(999)]),
              init:stop().
